homeassistant/components/reolink/__init__.py

```diff
@@ -0,0 +1,108 @@
+"""Reolink integration for HomeAssistant."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import cast
+
+import async_timeout
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    DEVICE_CONFIG_UPDATE_COORDINATOR,
+    DEVICE_UPDATE_INTERVAL,
+    DOMAIN,
+    HOST,
+    PLATFORMS,
+    SERVICE_PTZ_CONTROL,
+    SERVICE_SET_BACKLIGHT,
+    SERVICE_SET_DAYNIGHT,
+    SERVICE_SET_SENSITIVITY,
+)
+from .host import ReolinkHost
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Reolink from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    host = ReolinkHost(hass, cast(dict, entry.data), cast(dict, entry.options))
```

Comment:
```suggestion
    host = ReolinkHost(hass, dict(entry.data), dict(entry.options))
```

If it needs a `dict` we should convert it to dict or the typing change change in `ReolinkHost` to accept the types for `entry.data` and `entry.options`